core/block.go

```diff
@@ -85,7 +85,34 @@ func getBlockHashMetaInfo(network utils.Network) *blockHashMetaInfo {
 
 // VerifyBlockHash verifies the block hash. Due to bugs in Starknet alpha, not all blocks have
 // verifiable hashes.
-func VerifyBlockHash(b *Block, network utils.Network) error {
+func VerifyBlockHash(b *Block, network utils.Network) []error {
+	if len(b.Transactions) != len(b.Receipts) {
+		return []error{
+			fmt.Errorf("len of transaction: %v do not match len of receipts: %v",
+				len(b.Transactions), len(b.Receipts)),
+		}
+	}
+
+	for i, tx := range b.Transactions {
+		if !tx.Hash().Equal(b.Receipts[i].TransactionHash) {
+			return []error{
+				fmt.Errorf(
+					"transaction hash (0x%v) at index: %v does not match receipt's hash (0x%v)",
+					tx.Hash().Text(16), i, b.Receipts[i].TransactionHash),
+			}
+		}
+	}
+
+	var errs []error
```

Comment:
Would making this a chain of errors by wrapping make sense?

Reply:
at sync, we can follow the chain and report the "transaction hash failed" ones.

Reply:
Yeah, that is a good idea. I will think about how I can do that

Reply:
You this make me think we should wrap close errors as well instead of prioritising